#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE_NAME = "squirrel_navigation"

gen = ParameterGenerator()

gen.add("enabled", bool_t, 0, "Whether to use this plugin or not", True)

# Inflation parameters
gen.add("cost_scaling_factor", double_t, 0, "A scaling factor to apply to cost values during inflation.", 10, 0, 100)
gen.add("inflation_radius", double_t, 0, "The radius in meters to which the map inflates obstacle cost values.", 0.22, 0, 50)
gen.add("robot_link_radius", double_t, 0, "Radius of the robot link", 0.22, 0, 1.0)

exit(gen.generate(PACKAGE_NAME, "squirrel_navigation", "MapLayerPlugin"))